streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1331,6 +1307,20 @@ private void validateRackAwarenessConfiguration() {
         });
     }
 
+    private static void validateProcessingConfiguration(final String processingModeConfig) {
+        if (processingModeConfig.equals(EXACTLY_ONCE)) {
```

Comment:
I am wondering if we should just remove `StreamsConfig.EXACTLY_ONCE` and `StreamsConfig.EXACTLY_ONCE_BETA` and not throw a specific `ConfigException` those value, but just a generic one. When would we remove the specific `ConfigException` in the future? 
We had a deprecation period, we have the removal documented in the docs, I think that should be enough.

Reply:
Both solutions are fine with me. I suppose its a trade-off between a courtesy to the users and a cleaner codebase. @guozhangwang wdyt?

Reply:
My personal preference: even if we remove the deprecated strings, I still slightly preferred to log a meaningful message to users indicating that those config values are not used anymore, with the cost that we need to have those hard-coded values forever in the codebase. 

But in either way, I think we can still throw a `ConfigException` but a generic exception, what's the motivation for not doing so? @cadonna

Reply:
@guozhangwang My proposal was to throw a generic `ConfigException` instead of a specific one. I have never intended not to throw a `ConfigException`.

Reply:
Ah I see, thanks @cadonna .